cli/src/lib.rs

```diff
@@ -280,6 +282,9 @@ pub enum Command {
         program_id: Pubkey,
         /// Filepath to the new program binary.
         program_filepath: String,
+        /// Max times to retry on failure.
+        #[clap(long, default_value = "0")]
+        max_retries: u32,
```

Comment:
Doesn't `--max-sign-attempts` take care of this? Any failure other than blockhash expiration is likely to fail again e.g. not enough balance.

Reply:
In my experience, there's 2 common issues that sometimes just require another attempt: expired blockhash, failed write transactions. Occasionally, I've also seen a very uncommon but not nonexistent issue like a 503 from my RPC provider, or other even rarer things I can't quite remember but that went away with a retry.

Reply:
Here's a recording using this feature with the anchor program that inspired me to open this PR. I got "lucky" here and hit the trifecta: invalid blockhash, failed write transactions, and a new error I'm not actually sure I've seen before, that went away on its own.

https://github.com/user-attachments/assets/c33d135a-0679-4326-b4dd-b4303d1908f7